scripts/datastore_upgrade.py

```diff
@@ -0,0 +1,311 @@
+
+# The location on the local filesystem where Cassandra is installed.
+CASSANDRA_DIR = "/opt/cassandra"
+
+# The location of the Cassandra binary on the local filesystem.
+CASSANDRA_EXECUTABLE = CASSANDRA_DIR + "/cassandra/bin/cassandra"
+
+PID_FILE = "/var/appscale/appscale-cassandra.pid"
+
+# The default port to connect to ZooKeeper.
+ZK_PORT = ":2181"
+
+# The location on the local filesystem where we should store ZooKeeper data.
+ZK_DATA_LOCATION = "/opt/appscale/zookeeper"
+
+# The maximum number of entities to be returned from the query.
+_MAX_ENTITIES = 1000000
+
+def ensure_app_is_not_running():
+    """ Ensures AppScale is not running as this is an offline script. """
+    print ("Ensure AppScale is not currently running...")
+    appscale_running = subprocess.call(['service', CONTROLLER_SERVICE, 'status']) == 0
+    if appscale_running:
+        print ("AppScale is running, please shut it down and try again.")
+        exit(1)
+
+def start_cassandra():
+    """ Creates a monit configuration file and prompts Monit to start Cassandra. """
+    print ("Starting Cassandra...")
+    watch = "cassandra"
+    start_cmd = CASSANDRA_EXECUTABLE + " start -p " + PID_FILE
+    stop_cmd = "/usr/bin/python2 " + APPSCALE_HOME + "/scripts/stop_service.py java cassandra"
+    monit_interface.create_monit_config_file(watch, start_cmd, stop_cmd,
+        ports=[9999], match_cmd=CASSANDRA_DIR)
+    if not monit_interface.start(watch):
+        print ("Unable to start Cassandra.")
+        exit(1)
+    else:
+        print ("Successfully started Cassandra.")
+
+def start_zookeeper():
+    """ Creates a monit configuration file and prompts Monit to start ZooKeeper. """
+    print ("Starting ZooKeeper...")
+    os.system("rm -rfv /var/lib/zookeeper")
+    os.system("rm -rfv {}".format(ZK_DATA_LOCATION))
+    zk_server = "zookeeper-server"
+    if os.system("service --status-all|grep zookeeper$"):
+        zk_server = "zookeeper"
+
+    if not os.path.isdir(ZK_DATA_LOCATION):
+        print ("Initializing ZooKeeper")
+        os.system("/usr/sbin/service " + "zookeeper" + " stop")
+        os.system("mkdir -pv " + ZK_DATA_LOCATION)
+        os.system("chown -Rv zookeeper:zookeeper " + ZK_DATA_LOCATION)
+
+        if zk_server == "zookeeper-server":
+            if not os.system("/usr/sbin/service " + "zookeeper" + " init"):
+                print ("Failed to start zookeeper!")
+                exit(1)
+    os.system("ln -sfv /etc/zookeeper/conf/myid " + ZK_DATA_LOCATION + "/myid")
+    watch = "zookeeper"
+    start_cmd = "/usr/sbin/service " + "zookeeper" + " start"
+    stop_cmd = "/usr/sbin/service " + "zookeeper" + " stop"
+    match_cmd = "org.apache.zookeeper.server.quorum.QuorumPeerMain"
+    monit_interface.create_monit_config_file(watch, start_cmd, stop_cmd,
+        ports=[2181], match_cmd=match_cmd)
+    if not monit_interface.start(watch):
+        print ("Unable to start ZooKeeper")
+        exit(1)
```

Comment:
While they are pretty similar, `sys.exit` is recommended for use in programs while `exit` is mainly meant as a convenience function for the interactive shell.